akkeeper/src/main/scala/akkeeper/master/service/MasterService.scala

```diff
@@ -135,16 +135,19 @@ private[akkeeper] class MasterService(deployClient: DeployClient.Async,
       stopServicesWithError()
   }
 
-  private def initializedReceive: Receive = {
-    apiReceive orElse serviceTerminatedReceive
+  // Initial state. Used to receive messages from MonitoringService to join the cluster.
+  private def uninitializedReceive: Receive = {
+    fetchInstancesListReceive orElse serviceTerminatedReceive orElse apiStashReceive
   }
 
+  // Temporary state. Used while awaiting MemberUp response from the cluster leader.
```

Comment:
Nitpick: `MemberUp` is rather an `event generated by a cluster leader` than a 'response' (since there is no 'request')